nautobot/circuits/filters.py

```diff
@@ -46,13 +47,12 @@ class ProviderFilterSet(NautobotFilterSet):
     site_id = django_filters.ModelMultipleChoiceFilter(
         field_name="circuits__terminations__site",
         queryset=Site.objects.all(),
-        label="Site",
+        label="Site (ID) - Deprecated (use site filter)",
     )
-    site = django_filters.ModelMultipleChoiceFilter(
-        field_name="circuits__terminations__site__slug",
+    site = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="circuits__terminations__site",
```

Comment:
Is this filter tested/confirmed to work with this change? I can't remember if we have other cases where we're using `NaturalKeyOrPKMultipleChoiceFilter` with a nested lookup like this.

Reply:
Yup, Its been tested.